sdk/samples/iot/hub/src/sample_pnp_thermostat_component.c

```diff
@@ -0,0 +1,368 @@
+  AZ_RETURN_IF_FAILED(az_json_writer_init(&json_builder, payload, NULL));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_begin_object(&json_builder));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_property_name(&json_builder, telemetry_name));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_double(
+      &json_builder, handle->current_temperature, DOUBLE_DECIMAL_PLACE_DIGITS));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_end_object(&json_builder));
+  *out_payload = az_json_writer_get_bytes_used_in_destination(&json_builder);
+
+  return AZ_OK;
+}
+
+static az_result append_double(az_json_writer* json_writer, void* context)
+{
+  double value = *(double*)context;
+
+  return az_json_writer_append_double(json_writer, value, DOUBLE_DECIMAL_PLACE_DIGITS);
+}
+
+az_result sample_pnp_thermostat_init(
+    sample_pnp_thermostat_component* handle,
+    az_span component_name,
+    double initial_temp)
+{
+  if (handle == NULL)
+  {
+    return AZ_ERROR_ARG;
+  }
+
+  handle->component_name = component_name;
+  handle->current_temperature = initial_temp;
+  handle->min_temperature = initial_temp;
+  handle->max_temperature = initial_temp;
+  handle->device_temperature_avg_count = 1;
+  handle->device_temperature_avg_total = initial_temp;
+  handle->avg_temperature = initial_temp;
+  handle->send_max_temp_property = true;
+
+  return AZ_OK;
+}
+
+bool sample_pnp_thermostat_get_max_temp_report(
+    az_iot_hub_client* client,
+    sample_pnp_thermostat_component* handle,
+    sample_pnp_mqtt_message* mqtt_message)
+{
+  az_result result;
+  if (handle->send_max_temp_property)
+  {
+    if ((result = pnp_helper_create_reported_property(
+             mqtt_message->payload_span,
+             handle->component_name,
+             max_temp_reported_property_name,
+             append_double,
+             &handle->max_temperature,
+             &mqtt_message->out_payload_span))
+        != AZ_OK)
+    {
+      printf("Could not get reported property: error code = 0x%08x\n", result);
+      return false;
+    }
+    else if (
+        (result = az_iot_hub_client_twin_patch_get_publish_topic(
+             client, get_request_id(), mqtt_message->topic, mqtt_message->topic_length, NULL))
+        != AZ_OK)
+    {
+      printf("Error to get reported property topic with status: error code = 0x%08x\n", result);
+      return false;
+    }
+
+    handle->send_max_temp_property = false;
+
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+az_result sample_pnp_thermostat_get_telemetry_message(
+    az_iot_hub_client* client,
+    sample_pnp_thermostat_component* handle,
+    sample_pnp_mqtt_message* mqtt_message)
+{
+  az_result result;
+  if (az_failed(
+          result = pnp_helper_get_telemetry_topic(
+              client,
+              NULL,
+              handle->component_name,
+              mqtt_message->topic,
+              mqtt_message->topic_length,
+              NULL)))
+  {
+    return result;
+  }
+
+  if (az_failed(
```

Comment:
>az_failed [](start = 6, length = 9)

same

Reply:
wouldn't have the log message printed if we did